torch/csrc/jit/fuser/compiler.cpp

```diff
@@ -144,6 +156,111 @@ static void setInputBroadcastGroups(KernelSpec& spec) {
       std::back_inserter(spec.inputBroadcastGroups()));
 }
 
+// This function moves _grad_sum_to_size nodes along the computation graph
+// of the fusion group to the outputs and then records the shape inputs
+// in order for summation to be applied after the kernel.
+// Note that the correctness relies on the invariant that
+// _grad_sum_to_size is only applied to gradient nodes created by autodiff.
+// This is important because it ensures that in the mul and div nodes only
+// one argument (in the case of diff the numerator) has a summed value.
+// If two arguments to mul had one, we would be in trouble, but thanks
+// to the chain rule, we're OK.
+void processGradSumToSize(KernelSpec& spec) {
+  auto graph = spec.graph();
+
+  // keep track of one sumToSize hitting each output
```

Comment:
nit: "of _the last_ sumToSize on the path to an output". There's _at most_ one, so this sentence might not be true

Reply:
another nit -- this is not used until the very end of this function. Maybe move it there?